app/src/main/java/sopt/motivoo/presentation/exercise/ExerciseFragment.kt

```diff
@@ -79,7 +79,9 @@ class ExerciseFragment : BindingFragment<FragmentExerciseBinding>(R.layout.fragm
 
     private fun initAdapter(exerciseData: ExerciseData) {
         val adapter = ExerciseAdapter(userType = exerciseData.userType)
-        adapter.updateItemList(exerciseList = exerciseData.exerciseItemInfoList)
+        exerciseViewModel.exerciseHistoryInfoList.observe(binding.lifecycleOwner!!) {
+            adapter.submitList(it.toMutableList())
+        }
```

Comment:
```suggestion
      adapter.submitList(exerciseData.exerciseItemInfoList)
```

이미 ExerciseData를 가지고 있는데 이렇게 하면 안되나요??

Reply:
우선 해당 로직은 관찰하는 로직이므로 따로 빼서 observeLiveData에 넣어주는 것이 일관성 있어보입니다 !

Reply:
그리고 준희님 말씀처럼 exerciseHistoryInfoList를 관찰해야 할까요!? 요건 저도 정확히 모르니 설명해주시면 감사하겠습니다 !

Reply:
추가적으로 Fragment에서 관찰하실 때에는 viewLifeCycleOwner를 사용해주는 것이 좋습니다. 그 이유까지 알아보면 더 좋을 것 같아요 ! 찾다가 잘 모르시겠으면 질문해주세요 ~~